code/modules/clothing/glasses/glasses.dm

```diff
@@ -59,10 +59,13 @@
 
 ///Toggle the functions of the glasses
 /obj/item/clothing/glasses/proc/activate(mob/user)
-	if(activation_sound && deactivation_sound)
-		playsound(get_turf(src), active ? deactivation_sound : activation_sound, 15)
-
 	active = !active
+
+	if(active && activation_sound)
+		playsound(get_turf(src), activation_sound, 15)
+	else if(deactivation_sound)
+		playsound(get_turf(src), deactivation_sound, 15)
```

Comment:
if you are activating, and have no activation sound, it will play the deactivation sound.